pkgs/build-support/rust/default.nix

```diff
@@ -36,21 +37,22 @@ let
       cargoDepsCopy="$sourceRoot/${cargoVendorDir}"
     '';
 
+  ccForBuild="${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}cc";
+  cxxForBuild="${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}cxx";
+  ccForHost="${stdenv.cc.targetPrefix}cc";
+  cxxForHost="${stdenv.cc.targetPrefix}cxx";
+  releaseDir = "target/${stdenv.hostPlatform.config}/release";
+
 in stdenv.mkDerivation (args // {
   inherit cargoDeps;
 
   patchRegistryDeps = ./patch-registry-deps;
 
-  buildInputs = [ cacert git cargo rustc ] ++ buildInputs;
+  nativeBuildInputs = [ cargo rustc git cacert ] ++ nativeBuildInputs;
```

Comment:
Note these would be `buildPackges.cargo` `buildPackages.rustc`, etc, due to splicing black magic.